scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/DTLSSession.java

```diff
@@ -234,8 +240,26 @@ public SessionId getSessionIdentifier() {
 		return sessionIdentifier;
 	}
 
-	void setSessionIdentifier(SessionId sessionIdentifier) {
+	synchronized void setSessionIdentifier(SessionId sessionIdentifier) {
 		this.sessionIdentifier = sessionIdentifier;
+		this.connectionReadContext = null;
+		this.connectionWriteContext = null;
+	}
+
+	public synchronized EndpointContext getConnectionWriteContext() {
+		if (null == connectionWriteContext) {
+			connectionWriteContext = new DtlsEndpointContext(peer, peerIdentity, sessionIdentifier.toString(),
+					String.valueOf(writeEpoch), cipherSuite.name());
+		}
+		return connectionWriteContext;
```

Comment:
why not just recreate it dynamically ?

Reply:
FMPOV if something is frequently (re-) used, I prefer to keep it instead of dynamically recreate it. This may depend also on the amount of memory used, but though this seems to be a wrapper around already available information, that amount should not be an issue.

Reply:
Ok, personally I found the code a bit more complex and I'm not sure this optimization is really needed, but if you think this is better  let's go like this.

Reply:
@sophokles73 
Do you have also an opinion on this?

Reply:
(Just to be clear it's not a strong opinion)